plugins/binanceExchange_ws.go

```diff
@@ -180,6 +262,98 @@ func subcribeTicker(symbol string, state *mapEvents) (*stream, error) {
 
 }
 
+func subcribeUserStream(listenKey string, state *mapEvents) (*stream, error) {
+
+	userStreamLock := sync.Mutex{}
+
+	wsUserStreamExecutinReportHandler := func(message []byte) {
+
+		event := &eventExecutionReport{}
+		err := json.Unmarshal(message, event)
+
+		if err != nil {
+			log.Printf("Error unmarshal %s to eventExecutionReport\n", string(message))
+			return
+		}
+
+		userStreamLock.Lock()
+		defer userStreamLock.Unlock()
+
+		history, isHistory := state.Get(event.Symbol)
+
+		if !isHistory {
+			history.data = make(History, 0)
+			state.Set(event.Symbol, history)
+		}
+
+		now := time.Now()
+		history.createdAt = now
+
+		data, isOk := history.data.(History)
+
+		if !isOk {
+			log.Printf("Error conversion %v\n", ErrConversionHistory)
```

Comment:
should this also cause the stream to fail and throw an error somewhere?

I'm concerned that all these errors here will be hidden errors that will not cause the code to crash.

we want the bot to crash rather than throw silent errors

Reply:
one option it's to panic when I get errors from the stream

Reply:
See above comment. Once we agree on a pattern we can apply to all streams.